components/p2p/component.go

```diff
@@ -101,108 +123,199 @@ func provide(c *dig.Container) error {
+			}
+
+			peerAdded = true
+		}
+
+		p2pConfigManager.StoreOnChange(true)
+
+		if peerAdded {
+			if err := p2pConfigManager.Store(); err != nil {
+				Component.LogWarnf("failed to store peering config: %s", err)
+			}
 		}
 
-		dbKey, err := peerDB.LocalPrivateKey()
+		return p2pConfigManager
+	}); err != nil {
+		Component.LogPanic(err)
+	}
+
+	type p2pDeps struct {
+		dig.In
+		DatabaseEngine        hivedb.Engine `name:"databaseEngine"`
+		P2PDatabasePath       string        `name:"p2pDatabasePath"`
+		P2PBindMultiAddresses []string      `name:"p2pBindMultiAddresses"`
+	}
+
+	type p2pResult struct {
+		dig.Out
+		NodePrivateKey crypto.PrivKey `name:"nodePrivateKey"`
+		Host           host.Host
+	}
+
+	if err := c.Provide(func(deps p2pDeps) p2pResult {
+
+		res := p2pResult{}
+
+		privKeyFilePath := filepath.Join(deps.P2PDatabasePath, "identity.key")
+
+		// make sure nobody copies around the peer store since it contains the private key of the node
+		Component.LogInfof(`WARNING: never share your "%s" folder as it contains your node's private key!`, deps.P2PDatabasePath)
+
+		// load up the previously generated identity or create a new one
+		nodePrivateKey, newlyCreated, err := hivep2p.LoadOrCreateIdentityPrivateKey(privKeyFilePath, ParamsP2P.IdentityPrivateKey)
 		if err != nil {
-			Component.LogFatalAndExit(err)
+			Component.LogPanic(err)
 		}
-		libp2pPrivateKey, err := libp2putil.ToLibp2pPrivateKey(dbKey)
+		res.NodePrivateKey = nodePrivateKey
+
+		if newlyCreated {
+			Component.LogInfof(`stored new private key for peer identity under "%s"`, privKeyFilePath)
+		} else {
+			Component.LogInfof(`loaded existing private key for peer identity from "%s"`, privKeyFilePath)
+		}
+
+		peeringIP, err := readPeerIP()
 		if err != nil {
 			Component.LogFatalAndExit(err)
 		}
 
-		return peerOut{
-			PeerDB:         peerDB,
-			PeerDBKVSTore:  peerDBKVStore,
-			NodePrivateKey: libp2pPrivateKey,
+		if !peeringIP.IsGlobalUnicast() {
+			Component.LogWarnf("IP is not a global unicast address: %s", peeringIP)
 		}
-	}); err != nil {
-		return err
-	}
-
-	if err := c.Provide(func(nodePrivateKey crypto.PrivKey) host.Host {
-		var err error
 
 		connManager, err := connmgr.NewConnManager(
 			ParamsP2P.ConnectionManager.LowWatermark,
 			ParamsP2P.ConnectionManager.HighWatermark,
 			connmgr.WithGracePeriod(time.Minute),
 		)
 		if err != nil {
-			Component.LogErrorfAndExit("unable to initialize connection manager: %s", err)
+			Component.LogPanicf("unable to initialize connection manager: %s", err)
 		}
 
-		libp2pHost, err := golibp2p.New(
-			golibp2p.ListenAddrStrings(ParamsP2P.BindAddress),
-			golibp2p.Identity(nodePrivateKey),
-			golibp2p.ConnectionManager(connManager),
-			golibp2p.NATPortMap(),
+		// also see "defaults.go" in go-libp2p for the default values
+		createdHost, err := libp2p.New(
+			libp2p.ListenAddrStrings(ParamsP2P.BindMultiAddresses...),
+			libp2p.Identity(nodePrivateKey),
+			libp2p.Transport(tcp.NewTCPTransport),
```

Comment:
Why do we need this?

Reply:
To prevent the usage of the other default transport mechanisms like udp/websocket/quic.
Our tooling like the node-docker-setup for example or the wiki explains to open the port for TCP only anyways.